connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/outbound/DefaultOutboundConnectorFactory.java

```diff
@@ -49,32 +48,32 @@ public DefaultOutboundConnectorFactory() {
 
   @Override
   public List<OutboundConnectorConfiguration> getConfigurations() {
-    return new ArrayList<>(functionMap.keySet());
+    return new ArrayList<>(functionMap.values());
   }
 
   @Override
   public OutboundConnectorFunction getInstance(String type) {
-    var configuration =
-        functionMap.keySet().stream()
-            .filter(config -> config.type().equals(type))
-            .findFirst()
-            .orElseThrow(
-                () -> new NoSuchElementException("Connector " + type + " is not registered"));
-    return functionMap.get(configuration);
+    return Optional.ofNullable(functionMap.get(type))
+        .map(this::createInstance)
+        .orElseThrow(() -> new NoSuchElementException("Connector " + type + " is not registered"));
+  }
```

Comment:
Why not keep the previous mechanism of instance caching? Or is it to support a case where someone creates a stateful `OutboundConnectorFunction` and wants to use it with prototype scope in Spring?